src/coffee/delivery-export.coffee

```diff
@@ -0,0 +1,147 @@
+{SphereClient} = require 'sphere-node-sdk'
+Promise = require 'bluebird'
+_ = require 'underscore'
+Csv = require 'csv'
+objectPath = require 'object-path'
+
+class DeliveryExport
+  constructor: (@options, @logger) ->
+    @client = new SphereClient @options.client
+    @stats = @_getDefaultStats()
+
+    @options.export = _.defaults @options.export or {},
+      where: ''
+      perPage: 500
+
+  _getDefaultStats: ->
+    _.clone
+      loadedOrders: 0
+      exportedDeliveries: 0
+      exportedRows: 0
+
+  _getCsvHeader: ->
+    [
+      'orderNumber',
+      'delivery.id',
+      '_itemGroupId',
+      'item.id',
+      'item.quantity',
+      'parcel.id',
+      'parcel.length',
+      'parcel.height',
+      'parcel.width',
+      'parcel.weight',
+      'parcel.trackingId',
+      'parcel.carrier',
+      'parcel.provider',
+      'parcel.providerTransaction',
+      'parcel.isReturn'
+    ].join(',')
+
+  _castRowsToCsv: (rows) ->
+    new Promise (resolve, reject) ->
+      Csv().from rows
+        .on 'error', reject
+        .to.string resolve
+
+  # take delivery objects and map them to CSV rows
+  _mapDeliveriesToRows: (orderNumber, deliveries) ->
+    @stats.exportedDeliveries += deliveries.length
+
+    # run through all deliveries
+    _.flatten deliveries.map (delivery) =>
+      # create CSV rows from delivery items, orderNumber and deliveryId
+      # Note that every delivery has at least one item
+      rows = delivery.items.map (_item, index) =>
+        item =
+          orderNumber: orderNumber
+          'delivery.id': delivery.id
+          # add a unique ID to every delivery.item
+          # (one delivery can have multiple items with the same lineItemId and quantity)
+          '_itemGroupId': index
+          'item.id': _item.id
+          'item.quantity': _item.quantity
+
+        # add info about parcels, if there are no parcels fill cells with empty strings
+        @_connectItemWithParcel(item, delivery.parcels[index] or {})
+
+      # if there are more parcels then items, duplicate last CSV row
```

Comment:
When there are only one item and multiple deliveries, this item will be duplicated across multiple rows and the import script will know that it should ignore all next occurrences because of a duplicate `_itemGroupId` field